Framework/PythonInterface/plugins/algorithms/AlignComponents.py

```diff
@@ -623,30 +627,37 @@ def _minimisation_func(self, x_0, wks_name, component, firstIndex, lastIndex):
         # calculate the fractional peak center deviations, then sum their absolute values
         return np.sum(np.abs((peaks_d - self.peaks_ref) / self.peaks_ref))
 
-    def _getFirstDetID(self, component):
-        """
-        recursive search to find first detID of a component
+    def _unique_name(self, component, component_info):
```

Comment:
Can you make this functions belong to the instrument? I would like to use them somewhere else

Reply:
Please make a Story for this functionality
```c++
vector<size_t> Instrument::detectorIDs(std::string componentName) const;
```